Source/Utils/Average.H

```diff
@@ -57,38 +52,145 @@ namespace Average{
-     *                       object, extracted from its
-     *                       <a href="https://amrex-codes.github.io/amrex/doxygen/classamrex_1_1MultiFab.html">MultiFab</a>
-     *                       , containing the data to be averaged
+     * \param[in,out] mf_out MultiFab containing the floating point data to be filled
+     *                       with cell-centered averages
+     * \param[in]     mf_in  MultiFab containing the floating point data to be averaged
+     * \param[in]     dcomp  offset for the fourth component of the Array4 object,
+     *                       extracted from its MultiFab, where the cell-centered
+     *                       averages will be stored
+     * \param[in]     scomp  optional offset for the fourth component of the Array4
+     *                       object, extracted from its MultiFab, containing the data
+     *                       to be averaged
      * \param[in]     ncomp  optional number of components to loop over for both the
-     *                       input and the output
-     *                       <a href="https://amrex-codes.github.io/amrex/doxygen/structamrex_1_1Array4.html">Array4</a>
-     *                       objects, extracted from their respective
-     *                       <a href="https://amrex-codes.github.io/amrex/doxygen/classamrex_1_1MultiFab.html">MultiFab</a>
+     *                       input and the output Array4 objects, extracted from their
+     *                       respective MultiFab
      */
     void ToCellCenter ( MultiFab& mf_out,
                         const MultiFab& mf_in,
                         const int dcomp,
                         const int ngrow,
                         const int scomp=0,
                         const int ncomp=1 );
+
+    /**
+     * \brief Interpolates the floating point data contained in the input Array4
+     *        \c mf_fp_arr, extracted from a fine MultiFab, by averaging over either
+     *        1 point or 2 equally distant points.
+     *
+     * \param[in] mf_fp_arr floating point data to be interpolated
+     * \param[in] sf        staggering of the input fine MultiFab
+     * \param[in] sc        staggering of the output coarsened MultiFab
+     * \param[in] cr        coarsening ratio along each spatial direction
+     * \param[in] i         index along x of the coarsened Array4 to be filled
+     * \param[in] j         index along y of the coarsened Array4 to be filled
+     * \param[in] k         index along z of the coarsened Array4 to be filled
+     * \param[in] comp      index along the fourth component of the Array4 \c mf_fp_arr
+     *                      containing the data to be interpolated
+     *
+     * \return interpolated field at cell (i,j,k) of a coarsened Array4
+     */
+    AMREX_GPU_HOST_DEVICE
+    AMREX_FORCE_INLINE
+    Real CoarsenAndInterpolateKernel ( Array4<Real const> const& mf_fp_arr,
```

Comment:
This is not a MultiFab, and maybe including `src` in the name would help, e.g.,
```suggestion
    Real CoarsenAndInterpolateKernel ( Array4<Real const> const& src_arr,
```

Reply:
Yes, I can do that. The reason for calling this argument `mf_fp_arr` (and similar variables in `Source/Utils/Average.cpp`) was to make clear by the name itself what is the `MultiFab` (`mf_fp` in this case) whose data (box-wise) are being pointed to by that `Array4`. I would try anyways to be consistent, hence choose something like `mf_src`, `arr_src` together with `mf_dst` and `arr_dst`.